### PR TITLE
[MAINT] simplify tests decoding

### LINKED ISSUE #3570: simplify "complex" tests
list of "long" tests (function length > 50 lines) that are likely to contain too many asserts and could suffer "test pollution" and should probably be refactored.

```bash
$ flake8 nilearn/**/tests/test_*.py --select CFQ001 --max-function-length 60 
```

> [!TIP]
> # **💻 🍁  For the [hacktoberfest](https://hacktoberfest.com/) contributors 🍁 💻**
>
> Make sure to:
> - read our [contributing guidelines](https://nilearn.github.io/stable/development.html#contribution-guidelines)
>
> If this is your first time contributing to Nilearn, add your details in the [CITATION.cff file](https://github.com/nilearn/nilearn/blob/dcd92aa357ab3a751161aeea2edfa3a775ae644f/CITATION.cff#L42).
>
> For your pull request to be valid, you should refactor some of the long tests mentioned in the list below.
>

- [ ] nilearn/datasets/tests/test_atlas.py:445:1: CFQ001 Function test_fetch_atlas_yeo_2011 has length 63 that exceeds max allowed length 60
- [ ] nilearn/datasets/tests/test_func.py:668:1: CFQ001 Function test_fetch_development_fmri has length 63 that exceeds max allowed length 60
- [ ] nilearn/datasets/tests/test_neurovault.py:792:1: CFQ001 Function test_fetch_neurovault_ids has length 80 that exceeds max allowed length 60
- [ ] nilearn/decoding/tests/test_decoder.py:1442:1: CFQ001 Function test_decoder_vs_sklearn has length 70 that exceeds max allowed length 60
- [ ] nilearn/maskers/tests/test_surface_labels_masker.py:356:1: CFQ001 Function test_surface_label_masker_check_output_2d has length 72 that exceeds max allowed length 60
- [ ] nilearn/plotting/tests/test_find_cuts.py:143:1: CFQ001 Function test_find_cut_slices has length 65 that exceeds max allowed length 60
- [ ] nilearn/plotting/tests/test_find_cuts.py:284:1: CFQ001 Function test_find_parcellation_cut_coords has length 83 that exceeds max allowed length 60
- [ ] nilearn/plotting/tests/test_find_cuts.py:391:1: CFQ001 Function test_find_probabilistic_atlas_cut_coords has length 61 that exceeds max allowed length 60
- [ ] nilearn/plotting/tests/test_html_surface.py:27:1: CFQ001 Function test_get_vertexcolor has length 72 that exceeds max allowed length 60
- [ ] nilearn/plotting/tests/test_surf_plotting.py:1758:1: CFQ001 Function test_compute_facecolors_matplotlib has length 62 that exceeds max allowed length 60
- [ ] nilearn/regions/tests/test_signal_extraction.py:557:1: CFQ001 Function test_img_to_signals_labels_warnings has length 71 that exceeds max allowed length 60
- [ ] nilearn/reporting/tests/test_html_report.py:383:1: CFQ001 Function test_nifti_labels_masker_report has length 61 that exceeds max allowed length 60
- [ ] nilearn/tests/test_masking.py:781:1: CFQ001 Function test_extrapolate_out_mask has length 96 that exceeds max allowed length 60

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates to #3570

Changes proposed in this pull request:

- refactor tests of the decoding module to use setup functions and parametrize

### CODE REVIEW

File: nilearn/decoding/tests/test_decoder.py [hunk 1]
```diff
@@ -323,91 +353,47 @@ def test_decoder_dummy_classifier():
...
-        model = Decoder(estimator="svc", mask=mask, standardize=True, cv=cv)
-        if isinstance(cv, LeaveOneGroupOut):
-            groups = rand_local.binomial(2, 0.3, size=len(y))
-        else:
-            groups = None
-        model.fit(X, y, groups=groups)
-        assert accuracy_score(y, y_pred) > 0.9
+    assert model.scorer_._score_func == get_scorer("accuracy")._score_func
+    assert model.scorer_._sign == get_scorer("accuracy")._sign
+    assert model.score(X, y) > 0.5
```
Comment (bthirion) [inline nilearn/decoding/tests/test_decoder.py:396]: Looks like you're relaxing the test from .9 to .5 ?

**Review by bthirion — COMMENTED** [review]
LGTM otherwise ?

**Comment by Remi-Gau** [comment]
TODO

check that "similar" tests between modules have similar names and ideally similar "structure"

File: nilearn/decoding/tests/test_decoder.py [hunk 2]
```diff
@@ -323,91 +353,47 @@ def test_decoder_dummy_classifier():
...
     with pytest.raises(ValueError, match="'foo' is not a valid scoring value"):
         model.fit(X, y)
 
-    # Default scoring
-    model = Decoder(estimator="dummy_classifier", scoring=None)
-    assert model.scoring is None
-    model.fit(X, y)
-    assert model.scorer_._score_func == get_scorer("accuracy")._score_func
-    assert model.scorer_._sign == get_scorer("accuracy")._sign
-    assert model.score(X, y) > 0.5
```
Comment (Remi-Gau) [inline nilearn/decoding/tests/test_decoder.py:348]: The diff is not super clear but original expected score was 0.5

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Review by ymzayek — COMMENTED** [review]
LGTM! Do you plan on doing more in this PR? Otherwise I think can be merged

**Comment by Remi-Gau** [comment]
Nothing to add for now. So feel free to merge at will. 🚀

**Review by ymzayek — APPROVED** [review]

**Pull request merged**